packages/components/src/Dialog/Dialog.tsx

```diff
@@ -1,32 +1,84 @@
-import React from 'react';
+import React, { LegacyRef, RefObject } from 'react';
 import { ComponentProps } from '@marigold/types';
-
+import { useOverlayTriggerState } from '@react-stately/overlays';
+import { OverlayContainer } from '@react-aria/overlays';
+import { useButton } from '@react-aria/button';
 import { Close } from '@marigold/icons';
-import { Button } from '../Button';
+
 import { Box } from '../Box';
+import { Button } from '../Button';
+import { Heading } from '../Heading';
+
+import { ModalDialog } from './ModalDialog';
 
 export type DialogProps = {
-  onClose?: ComponentProps<typeof Button>['onClick'];
+  isOpen: boolean;
+  close: ComponentProps<typeof Button>['onClick'];
+  title?: string;
 } & ComponentProps<'div'>;
 
 export const Dialog: React.FC<DialogProps> = ({
-  onClose,
   children,
+  title,
   className,
+  isOpen = false,
+  close,
   ...props
 }) => {
+  const closeButtonRef = React.useRef<HTMLElement>() as RefObject<HTMLElement>;
+
+  // useButton ensures that focus management is handled correctly,
+  // across all browsers. Focus is restored to the button once the
+  // dialog closes.
+  const { buttonProps: closeButtonProps } = useButton(
+    {
+      onPress: () => close(),
+    },
+    closeButtonRef
+  );
+
   return (
-    <Box display="flex" width="100%">
-      <Box {...props} variant="dialog.wrapper" className={className}>
-        <Box display="flex">
-          <Box variant="dialog.body">{children}</Box>
+    <OverlayContainer>
+      <ModalDialog isOpen={isOpen} onClose={close} isDismissable>
+        <Box variant="dialog.wrapper" className={className} {...props}>
+          <Box variant="dialog.body">
+            {title && (
+              <Heading as="h4" variant="h4">
+                {title}
+              </Heading>
+            )}
+            {children}
+          </Box>
           <Box variant="dialog.onClose">
-            <Button variant="close" size="xsmall" onClick={onClose}>
+            <Button
+              variant="close"
+              size="xsmall"
+              {...closeButtonProps}
+              ref={closeButtonRef as LegacyRef<HTMLButtonElement>}
```

Comment:
I don't really understand what `LegacyRef` is or does, can you explain that?

Reply:
Oh I think I had needed that.... but is probably not necessary